sdks/java/core/src/main/java/org/apache/beam/sdk/providers/GenerateSequenceSchemaTransformProvider.java

```diff
@@ -0,0 +1,188 @@
+  public static final Schema OUTPUT_SCHEMA = Schema.builder().addInt64Field("value").build();
+
+  @Override
+  public String identifier() {
+    return "beam:schematransform:org.apache.beam:generate_sequence:v1";
+  }
+
+  @Override
+  public List<String> inputCollectionNames() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  public List<String> outputCollectionNames() {
+    return Collections.singletonList(OUTPUT_ROWS_TAG);
+  }
+
+  @Override
+  public Class<GenerateSequenceConfiguration> configurationClass() {
+    return GenerateSequenceConfiguration.class;
+  }
+
+  @Override
+  public SchemaTransform from(GenerateSequenceConfiguration configuration) {
+    return new GenerateSequenceSchemaTransform(configuration);
+  }
+
+  @DefaultSchema(AutoValueSchema.class)
+  @AutoValue
+  public abstract static class GenerateSequenceConfiguration {
+    @AutoValue
+    public abstract static class Rate {
+      @SchemaFieldDescription("Number of elements component of the rate.")
+      public abstract Long getElements();
+
+      @SchemaFieldDescription("Number of seconds component of the rate.")
+      @Nullable
+      public abstract Long getSeconds();
+
+      public static Builder builder() {
+        return new AutoValue_GenerateSequenceSchemaTransformProvider_GenerateSequenceConfiguration_Rate
+            .Builder();
+      }
+
+      @AutoValue.Builder
+      public abstract static class Builder {
+        public abstract Builder setElements(Long elements);
+
+        public abstract Builder setSeconds(Long seconds);
+
+        public abstract Rate build();
+      }
+    }
+
+    public static Builder builder() {
+      return new AutoValue_GenerateSequenceSchemaTransformProvider_GenerateSequenceConfiguration
+          .Builder();
+    }
+
+    @SchemaFieldDescription("The minimum number to generate (inclusive).")
+    public abstract Long getStart();
+
+    @SchemaFieldDescription(
+        "The maximum number to generate (exclusive). Will be an unbounded sequence if left unspecified.")
+    @Nullable
+    public abstract Long getEnd();
+
+    @SchemaFieldDescription(
+        "Specifies the rate to generate a given number of elements per a given number of seconds. Applicable only to unbounded sequences.")
+    @Nullable
+    public abstract Rate getRate();
+
+    @AutoValue.Builder
+    public abstract static class Builder {
+
+      public abstract Builder setStart(Long start);
+
+      public abstract Builder setEnd(Long end);
+
+      public abstract Builder setRate(Rate rate);
+
+      public abstract GenerateSequenceConfiguration build();
+    }
+
+    public void validate() {
+      checkNotNull(this.getStart(), "Must specify a starting point \"start\".");
+      Long start = this.getStart();
+      Long end = this.getEnd();
+      if (end != null) {
+        checkArgument(end == -1 || end >= start, "Invalid range [%s, %s)", start, end);
+      }
+      Rate rate = this.getRate();
+      if (rate != null) {
```

Comment:
This should validate that the rate itself isn't null as well, right?

Reply:
or is there a default rate?

Reply:
The `elements` component is required, so not setting it will result in an encoding error. The `seconds` component is nullable and defaults to `1L` in line 174